apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex

```diff
@@ -500,19 +500,19 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
            maybe_handle_cache_miss(maybe_descriptor, interface, new_state),
          :ok <- can_write_on_interface?(interface_descriptor),
          interface_id <- interface_descriptor.interface_id,
-         {:ok, endpoint} <- resolve_path(path, interface_descriptor, new_state.mappings),
-         endpoint_id <- endpoint.endpoint_id,
-         db_retention_policy = endpoint.database_retention_policy,
-         db_ttl = endpoint.database_retention_ttl,
+         {:ok, mapping} <- resolve_path(path, interface_descriptor, new_state.mappings),
+         endpoint_id <- mapping.endpoint_id,
```

Comment:
nit:

I know you're not modifying it but you're touching that line so,

this will always match, it's better to make it explicit

```suggestion
         endpoint_id = mapping.endpoint_id,
```